src/indra_cogex/client/queries.py

```diff
@@ -1105,14 +1105,36 @@ def _get_mesh_child_terms(
         to the graph.
     """
     meshid_norm = norm_id(*mesh_term)
+    # todo: figure out why [:isa|partof*1..] is ~170x faster than [:isa*1..]
+    #  for the query below
     query = (
         """
-        MATCH (c:BioEntity)-[:isa*1..]->(:BioEntity {id: "%s"})
-        RETURN DISTINCT c.id
+        MATCH p=(c:BioEntity)-[:isa|partof*1..]->(:BioEntity {id: "%s"})
+        RETURN p
     """
         % meshid_norm
     )
-    return set(client.query_tx(query, squeeze=True))
+    paths = client.query_tx(query, squeeze=True)
+    relations_list = [client.neo4j_to_relations(p) for p in paths]
+    mesh_ids = set()
+    for path_result in relations_list:
+        current_set = set()
+        for relation in path_result:
+            # Check if relation is 'isa' and if both nodes are mesh nodes
+            if relation.rel_type == "isa" and \
```

Comment:
If it's guaranteed that the other node in a `isa` relation from/to a mesh term node always is a mesh term, this filtering is unnecessary.

Reply:
I actually think we could omit all the extra logic here since both `isa` and `partof` are usually legitimate when in comes to finding child terms. In the case of MeSH, `partof` will not appear in practice so we could effectively keep the old code and expect it to work.